### ISSUE TITLE
Approve manual actions

### ISSUE THREAD

**Issue by rcollette** [nfriend/lab-assistant#2]
From our master branch, we gate deployments from dev to qa and from qa to production with a manual job trigger.   I would like to be able to execute these manual triggers from Alexa.   Ex " promote project xyz" 

Is there a way to do this currently with this app? If not does the GitLab API support execution of manual triggers such that this could be done reasonably from Alexa?.   I would be willing to contribute.

**Comment by nfriend** [comment]
@rcollette I think this _might_ be possible, depending on what you need.

Check out the [Pipeline variables](https://github.com/nfriend/lab-assistant#pipeline-variables) section of the README. If you can build your `.gitlab-ci.yml` in such a way that the "promote" jobs are regular jobs that only run when certain variables are populated, then I think it should be possible to do what you want with a job like this:

```yml
deploy_to_qa:
  script:
    - echo 'Deploying to qa....'
  only:
    variables:
      - $LAB_ASSISTANT && $LAB_ASSISTANT_TEST
```

This job would be triggered by a phrase like "Alexa, start a new test pipeline".

The caveat is that Lab Assistant always creates a _new_ pipeline. It can't trigger manual jobs on existing pipelines. So I think this is unlikely to work for most projects that use trunk-based deployments, since this would always deploy the _latest_ code to the target environment.

---

One other thought I had: if you are able to build a shell script that can perform the logic you want using the GitLab API, another approach is to have Alexa trigger a "dummy" job whose only job is to trigger the shell script, which then performs the logic you need.

**Comment by rcollette** [comment]
I would probably have to use the latter approach.   But I'm starting to wonder if this will be sufficiently useful for me.  Without notifications of a pipeline pending manual approval (as mentioned in my other issue), I would wind up having to be in gitlab to check status anyway.    I definitely don't want my pipeline coupled to Alexa with special "announcement" jobs or scripting since on my team I may be the only team using it.

**Comment by nfriend** [comment]
@rcollette Makes sense! I agree the lack of notification support is a huge missing feature.
